app/controllers/course/user_invitations_controller.rb

```diff
@@ -23,22 +23,29 @@ def authorize_invitation!
   def course_user_invitation_params # :nodoc:
     invitations_attributes = { course_user: [:name], user_email: [:email] }
     @invite_params ||= params.require(:course).
-                       permit(:invitations_file, invitations_attributes: invitations_attributes)
+                       permit(:invitations_file, :registration_key,
+                              invitations_attributes: invitations_attributes)
   end
 
   # Determines the parameters to be passed to the invitation service object.
   #
-  # @return [Hash]
+  # @return [Tempfile|Hash|String|nil]
   def invitation_params
-    course_user_invitation_params[:invitations_file].try(:tempfile) ||
-      course_user_invitation_params[:invitations_attributes]
+    @invitation_params ||= course_user_invitation_params[:invitations_file].try(:tempfile) ||
+                           course_user_invitation_params[:invitations_attributes] ||
+                           course_user_invitation_params[:registration_key]
   end
 
   # Invites the users via the service object.
   #
   # @return [bool] True if the invitation was successful.
   def invite
-    invitation_service.invite(invitation_params)
+    case invitation_params
```

Comment:
Is it better to use submit button names (e.g. ` f.button :submit, name: 'code_registration'`) instead of param types ?